src/fuselage/CCPACSDoorAssemblyPosition.cpp

```diff
@@ -16,17 +16,171 @@
 
 #include "CCPACSDoorAssemblyPosition.h"
 
+#include "CCPACSFuselage.h"
+#include "CCPACSFrame.h"
+#include "CCPACSFuselageStringer.h"
+#include "CTiglUIDManager.h"
+#include "CNamedShape.h"
+
+#include <TopoDS_Compound.hxx>
+#include <TopTools_IndexedMapOfShape.hxx>
+#include <TopExp.hxx>
+#include <TopoDS_Builder.hxx>
+#include <BRep_Tool.hxx>
+#include <BRepTools.hxx>
+#include <BRepExtrema_ExtCC.hxx>
+#include <Bnd_Box.hxx>
+#include <BRepBndLib.hxx>
+
+#include "tiglcommonfunctions.h"
+
 namespace tigl
 {
 CCPACSDoorAssemblyPosition::CCPACSDoorAssemblyPosition(CCPACSCargoDoorsAssembly* parent, CTiglUIDManager* uidMgr)
     : generated::CPACSDoorAssemblyPosition(parent, uidMgr)
-    , invalidated(true)
 {
 }
 
 void CCPACSDoorAssemblyPosition::Invalidate()
 {
-    invalidated = true;
+    m_borderCache = boost::none;
+    m_geometry    = boost::none;
+}
+
+TopoDS_Shape CCPACSDoorAssemblyPosition::GetGeometry(TiglCoordinateSystem referenceCS)
+{
+    if (!m_geometry)
+        BuildGeometry();
+
+    if (referenceCS == GLOBAL_COORDINATE_SYSTEM)
+        return m_parent->GetParent()->GetParent()->GetTransformationMatrix().Transform(m_geometry.value());
+    else
+        return m_geometry.value();
+}
+
+bool CCPACSDoorAssemblyPosition::Contains(const TopoDS_Face& face)
+{
+    if (!m_borderCache)
+        UpdateBorders();
+
+    // compute center point of face
+    double u_min = 0., u_max = 0., v_min = 0., v_max = 0.;
+    BRepTools::UVBounds(face, u_min, u_max, v_min, v_max);
+    Handle(Geom_Surface) testSurf = BRep_Tool::Surface(face);
+    const gp_Pnt center           = testSurf->Value(u_min + ((u_max - u_min) / 2), v_min + ((v_max - v_min) / 2));
+
+    return Contains(center);
+}
+
+bool CCPACSDoorAssemblyPosition::Contains(const gp_Pnt& point)
+{
+    const double _45DegInRad = Radians(45.0);
+
+    const BorderCache& c = m_borderCache.value();
+    gp_Ax1 test1(c.sFrame_sStringer.Location(), gp_Vec(c.sFrame_sStringer.Location(), point));
+    if (test1.Angle(c.sFrame_sStringer) < _45DegInRad) {
```

Comment:
Again a wood of braces. Seems to look very similar to the other code we already had. Can we reuse the other code?